### DISCUSSION TITLE
SwipeToRefresh feature

### DISCUSSION THREAD

**Discussion by N7ghtm4r3** [Ahmad-Hamwi/lazy-pagination-compose#7] [Ideas]
Hi, could be possibile a built-in integration of `swipe-to-refresh` gesture like the `firstPageProgressIndicator`, `newPageProgressIndicator`, `firstPageErrorIndicator` or `newPageErrorIndicator` utilities? Because it should be very useful to have

**Comment by Ahmad-Hamwi** [comment]
Hey, thank you for you suggestion. This would be out of the scope of this library, since it's solely focused on encapsulating pagination logic, but the good news is that you're going to still be able to implement your own swipe to refresh functionality as if you were dealing with a normal LazyColumn.

This is already implemented in the sample of this library, take a look the PaginatedLazyColumn [here](https://github.com/Ahmad-Hamwi/lazy-pagination-compose/blob/main/sample/composeApp/src/commonMain/kotlin/ui/lazyColumn/Content.kt).

**Comment by N7ghtm4r3** [comment]
Thank you for the info and tip! I'll check it to implement by myself

**Comment by N7ghtm4r3** [comment]
Hi, I am sorry for the following question, but I'm implementing your library gradually on my application. Is there a way to refresh the `pageState.allItems` list without invoke the `refresh` method like using a `SnapshotStateList` to add or remove locally the items and update the `UI`? 
For example to add an element making a request to the backend. Instead to refresh the list with the `refresh` method, but instead adding directly  to the `allItems` list updating the `UI` avoiding that invocation

**Reply by Ahmad-Hamwi** [reply]
Yes there is, check the latest version [v1.3.7](https://github.com/Ahmad-Hamwi/lazy-pagination-compose/releases/tag/v1.3.7). The function that you're looking for is called `appendPageWithUpdates`

**Reply by N7ghtm4r3** [reply]
Thank you again!
